src/Microsoft.DotNet.Build.Tasks.Feed/src/BlobArtifactModelFactory.cs

```
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.DotNet.VersionTools.BuildManifest.Model;
+
+namespace Microsoft.DotNet.Build.Tasks.Feed
+{
+    public interface IBlobArtifactModelFactory
+    {
+        BlobArtifactModel CreateBlobArtifactModel(ITaskItem item, TaskLoggingHelper log);
+    }
+
+    public class BlobArtifactModelFactory : IBlobArtifactModelFactory
+    {
+        public BlobArtifactModel CreateBlobArtifactModel(ITaskItem item, TaskLoggingHelper log)
+        {
+            string path = item.GetMetadata("RelativeBlobPath");
+            if (string.IsNullOrEmpty(path))
+            {
+                log.LogError($"Missing 'RelativeBlobPath' property on blob {item.ItemSpec}");
```

Was the odd behavior here before, where we log the error, but then just continue running anyway, returning what might be a suprizingly broken data model?

---

This existed before I refactored. I wonder if we continue in the event we find other errors that the user would need to mitigate?

---

After talking to Mitchell about this, I have a better understanding of what's going with these error logs: Since a lot of this code is currently used by MSBuild Tasks, we want to gather as many errors as possible and then output them to the user for them to correct as many errors as possible. The alternative, which is usually preferred, of failing early, would only present one error at a time to the user, which would then require them to re-run their build that uses the MSBuild Tasks, and that may take a long time to uncover the next error. While, not typically ideal, in this process, we gather as many errors as possible to return to the user. 

Likely, this won't be the only time someone wonders about this behavior, so I'm adding a comment on this code here. I'll try to add it to other places we do the same kind of functionality. If we find that we're starting to use this code for things other than MSBuild Tasks, perhaps we can investigate how to fail early with non-MSBuild functionality.